app/models/manageiq/providers/amazon/cloud_manager/event_catcher/stream.rb

```diff
@@ -110,19 +110,23 @@ def sqs_get_queue_url(queue_name)
     end
   end
 
-  # @return [Aws::SNS::Topic] the found topic
-  # @raise [ProviderUnreachable] in case the topic is not found
   def sns_topic
     @ems.with_provider_connection(:service => :SNS) do |sns|
-      sns.topics.detect { |t| t.arn.split(/:/)[-1] == @topic_name }
-    end || begin
-      $aws_log.warn("#{log_header} Unable to find the AWS Config Topic '#{@topic_name}'. " \
-      "Cannot collect Amazon events for AWS Access Key ID #{@ems.authentication_userid}")
-      $aws_log.warn("#{log_header} Contact Amazon to create the AWS Config service and topic for Amazon events.")
-      raise ProviderUnreachable, "Unable to find the AWS Config Topic '#{@topic_name}'"
+      get_topic(sns) || create_topic(sns)
     end
   end
 
+  def get_topic(sns)
+    sns.topics.detect { |t| t.arn.split(/:/)[-1] == @topic_name }
+  end
+
+  def create_topic(sns)
+    sns.create_topic(:name => @topic_name)
```

Comment:
@durandom - Wouldn't that mean the end user does not want event support? In which case the event role would be disabled in ManageIQ.
How would I programmatically determine that CloudWatch and AWS Config are disabled? I looked at the API documentation but didn't see anything to that end.